CONTRIBUTING.md

```diff
@@ -500,20 +500,37 @@ it is acceptable to follow the convention in that module; though in general, it
 would be preferable to follow the motto of "leave the codebase in better shape
 than you found it."
 
-## katana::Result
+## Error Codes and Messages
 
-On older compilers, auto conversion to `katana::Result` will fail for types
-that can't be copied. One symptom is compiler errors on GCC 7 but not on GCC 9.
-We've adopted the workaround of returning such objects like so:
+Code that needs to indicate an error to callers typically should use
+`katana::Result`. When a function returns `katana::Result<T>`, it means that
+the function either returns a `T` or an error. When a function returns
+`katana::Result<void>`, it means that either the function succeeds (i.e.,
+returns `void`) or returns an error.
 
+A common pattern for using a function that returns a `katana::Result<T>` is:
 ```cpp
-Result<Thing> MakeMoveOnlyThing() {
-  Thing t;
-  ....
-  return Thing(std::move(t));
+auto r = ReturnsAResult();
+if (!r) {
+  // Some error happened. Either...
+
+  // ... we handle it
+  if (r.error() == ErrorCode::Foo) {
+    continue;
```

Comment:
Don't forget `continue` is a keyword ans not valid here. Do you mean `// ... continue`, as in "code that implements handling goes here"?

Reply:
I updated the PR to something more believable.